htdocs/comm/action/class/api_agendaevents.class.php

```diff
@@ -114,14 +114,14 @@ function index($sortfield = "t.id", $sortorder = 'ASC', $limit = 100, $page = 0,
 
         // If the internal user must only see his customers, force searching by him
         $search_sale = 0;
-        if (! DolibarrApiAccess::$user->rights->societe->client->voir && !$socid) $search_sale = DolibarrApiAccess::$user->id;
+        if (! empty($conf->societe->enabled) && ! DolibarrApiAccess::$user->rights->societe->client->voir && !$socid) $search_sale = DolibarrApiAccess::$user->id;
 
         $sql = "SELECT t.id as rowid";
-        if ((!DolibarrApiAccess::$user->rights->societe->client->voir && !$socid) || $search_sale > 0) $sql .= ", sc.fk_soc, sc.fk_user"; // We need these fields in order to filter by sale (including the case where the user can only see his prospects)
```

Comment:
This change breaks the security.
What value of $search_sale do you have when you make your search ?

Reply:
@eldy 
by example, if i use only "agenda" and "member" module, i can't see action in agenda with the old test

Reply:
@eldy  you see ?

Reply:
I may have an idea of what's wrong.
Can you check if user used to make the api call is an external user or not (so property user->socid is defined)
If only module agenda and member is on, you may need to check in database, because you should not be able to see this in database.

Reply:
@eldy 
ok thank you